nms/app/packages/magmalte/app/views/subscriber/SubscriberTable.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+import type {ActionQuery} from '../../components/ActionTable';
+import type {TableProps} from './SubscriberOverview';
+import type {WithAlert} from '@fbcnms/ui/components/Alert/withAlert';
+
+import ActionTable from '../../components/ActionTable';
+import NetworkContext from '../../components/context/NetworkContext';
+import React from 'react';
+import SubscriberContext from '../../components/context/SubscriberContext';
+import nullthrows from '@fbcnms/util/nullthrows';
+import {JsonDialog} from './SubscriberOverview';
+
+import {FEG_LTE} from '@fbcnms/types/network';
+import {handleSubscriberQuery} from '../../state/lte/SubscriberState';
+import {useContext, useState} from 'react';
+import {useEnqueueSnackbar} from '@fbcnms/ui/hooks/useSnackbar';
+import {useRouter} from '@fbcnms/ui/hooks';
+
+export type SubscriberRowType = {
+  name: string,
+  imsi: string,
+  activeApns?: string,
+  ipAddresses?: string,
+  activeSessions?: number,
+  service: string,
+  currentUsage: string,
+  dailyAvg: string,
+  lastReportedTime: Date | string,
+};
+
+export default function SubscriberTable(props: WithAlert & TableProps) {
+  const {history, match, relativeUrl} = useRouter();
+  const [currRow, setCurrRow] = useState<SubscriberRowType>({});
+  const networkId: string = nullthrows(match.params.networkId);
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const ctx = useContext(SubscriberContext);
+  const networkCtx = useContext(NetworkContext);
+  const subscriberMetrics = ctx.metrics;
+  const [jsonDialog, setJsonDialog] = useState(false);
+  const [pageSize, setPageSize] = useState(10);
+  // first token (page 1) is an empty string
+  const [tokenList, setTokenList] = useState(['']);
```

Comment:
I'm apprehensive to using tokenList given that the tokens may be stale over time. We can just keep track of next page and previous page, perhaps when we refresh, we can clear the token list I guess

Reply:
If we're on page 4, we would keep track of previous page token (page 3) and next page token (page 5). We would not be able to go back to page 2